lib/src/source/hosted.dart

```diff
@@ -565,6 +578,11 @@ class BoundHostedSource extends CachedSource {
     } else if (error is io.TlsException) {
       fail('Got TLS error trying to find package $package at $url.', error,
           stackTrace);
+    } else if (error is FormatException) {
+      throw PackageNotFoundException(
+          'Got badly formatted response trying to find package $package at $url',
```

Comment:
Hmm, if `PUB_HOSTED_URL` is not the default value, then maybe we should include a message warning about that here -- it's common that people add an extra slash or something to their `PUB_HOSTED_URL` or they use `https://localhost...` when it should have been `http://localhost...`

That probably applies to `SocketException`, `TlsException` and `PubHttpException`...

Just an idea, that might be an orthogonal concern best addressed in a follow up PR.

Reply:
I think the url in itself might be enough.